File: DSA/LeetCode/Math/921. Minimum Add to Make Parentheses Valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:

        close=0 
        open=0
        

        for c in s:
            if c == "(":
                open +=1
            else:
                if open >0:
                    open -=1
                else:
                    close +=1

        return close +  open

File: DSA/LeetCode/Math/921. Minimum Add to Make Parentheses Valid.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function (s) {

    let open = 0
    let close = 0


    for (let i = 0; i < s.length; i++) {

        if (s[i] == "(") {
            open++

        } else if (s[i] == ")") {
            if (open > 0) {
                open--
            } else {
                close++
            }

        }

    }

    return close + open



};